src/I18n/I18n.php

```diff
@@ -215,7 +215,9 @@ public static function locale($locale = null)
 
         if (!empty($locale)) {
             Locale::setDefault($locale);
-            static::translators()->setLocale($locale);
+            if (isset(static::$_collection)) {
+                static::translators()->setLocale($locale);
```

Comment:
It is set here https://github.com/cakephp/cakephp/blob/3.0-lazy-translators/src/I18n/I18n.php#L69